osp/core/session/sparql_backend.py

```diff
@@ -52,6 +52,7 @@ class SparqlResult(ABC):
     def __init__(self, session):
```

Comment:
I would accept here also the datatypes as keyword argument with `None` as default value (and set this default if not `None` three lines below), otherwise `None` or an empty one. Good if those result objects need to be integrated with something else.